openedx_events/learning/signals.py

```diff
@@ -350,3 +353,51 @@
         "submission": ORASubmissionData,
     },
 )
+
+
+# .. event_type: org.openedx.learning.course.passing.status.updated.v1
+# .. event_name: COURSE_PASSING_STATUS_UPDATED
+# .. event_description: Emitted when course grade updates.
```

Comment:
Every time a course grade updates or when it upgrades to `passing`?

Reply:
Each time the user-course grade is updated: here we completely mirrored the behavior of the internal LMS events (COURSE_GRADE_NOW_PASSED, COURSE_GRADE_NOW_FAILED), but we unified both in a single event type with a payload attr `status`.

The same for the CCX analog event type - LMS just emits a different public signal for CCX due to a unique payload.

Reply:
I understand better now, thank you! I have a few comments about it:

When I read `COURSE_PASSING_STATUS_UPDATED,` I think about the status passing changing, something like `is_passing=True` or `is_passing=False`. But in reality, it could be either the course grade status changing to passing or failing. Is that correct? 

What do you think about using `COURSE_GRADE_STATUS_CHANGED` (or updated) instead? or using a boolean instead of a str to know the grade status?

Reply:
@mariajgrimaldi  we started these as separate events (so event's nature already includes the passing status, instead of carrying it in the payload), but it appeared that for example the `COURSE_GRADE_NOW_PASSED` is stateless, meaning it fires on any course grade update until user's grade is above course grading policy threshold value. So it is not about `...PASSED` but about `...PASSING` - there is no guarantee it has started being "passing" right now...

So it is definitely not about status `change` but more about `current status` - "after the last problem submission User's course grade value (became/still) is under/below course's grading policy".

We can split the single universal event into 2 dedicated events (and therefore rid of the `is_passing=True | status=passing` dilemma):
- COURSE_GRADE_IS_PASSING
- COURSE_GRADE_IS_FAILING
(better naming is appreciated, though)

> But the main idea is about the fact these events are meant **not about status transition** but rather **about continuous status**.

Reply:
@mariajgrimaldi TL;DR 

1. We will update the universal event names to `COURSE_GRADE_STATUS` and `CCX_COURSE_GRADE_STATUS`;
2. The new event types would be:
  - `org.openedx.learning.course.grade.status.v1`;
  - `org.openedx.learning.ccx.course.grade.status.v1`;
3. Payload will be updated: `status = passing | failing` >> `is_passing = <bool>`;

Is this something we are looking for?

Reply:
I see, thank you for the detailed explanation! I needed clarification with the description + the naming of the event, but I got it now. This event will be emitted every time to notify the passing status of the grade. So, the status doesn't necessarily need to change for this event to be sent, but an operation with the grade must happen. Could we then add more detail to the event description and clarify it? So we can leave it as is. Thank you! I am sorry for all the confusion.

Reply:
Could we update the description to reflect this discussion?